### PR TITLE
CotE - Miya library

### LINKED ISSUE #2792: CotE - Miya Library
ml

### PR DESCRIPTION
**Pull request by Matthew-Griffith**
closes #2792 

here is a rough draft of Miya Library. I didn't use HandlerMenuPrompt but thought that once it was merged I come easily convert the class to use that instead. I don't have a lot of test for it yet, I just wrote one for how I thought the card should work with the plan to write more once I got this one to pass.

### CODE REVIEW

**Comment by Matthew-Griffith** [comment]
oh I just had the thought that right now the card isn't actually checking that the card selected is imperial and a character. I'll add that tomorrow.

File: server/game/cards/06-CotE/MiyaLibrary.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    setupCardAbilities() {
+        this.action({
+            title: 'Replace Miya Library for a faceup imperial character',
+            condition: context => context.player.dynastyDeck.size() > 0,
+            effect: 'Search the top five card for your dynasty deck for an imperial character',
+            handler: context => {
+                this.game.promptWithHandlerMenu(context.player, {
+                    activePromptTitle: 'select an imperial character to replace miya library',
+                    context: context,
+                    cards: context.player.dynastyDeck.first(5),
```
[the remark this answers is not in the document]
Comment (Matthew-Griffith) [reply server/game/cards/06-CotE/MiyaLibrary.js]: Thanks for catching that. I don't know why I thought it was five.

File: server/game/cards/06-CotE/MiyaLibrary.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+                    cardHandler: (card) => {
+                        let choices = context.player.dynastyDeck.first(4);
+                        if(card.hasTrait('imperial') && card.getType() === CardTypes.Character) {
+                            context.player.moveCard(card, context.source.location);
+                            card.facedown = false;
+                            choices.splice(choices.indexOf(card), 1);
+                            choices.push(context.source);
+                        }
+                        this.miyaLibraryPrompt(context, choices, [], 'Select the card you would like to place on top of your dynasty deck');
+                    }
```
Comment (jeremylarner) [inline server/game/cards/06-CotE/MiyaLibrary.js]: I think this should be:
`cardHandler: (card) => {
                         if(card.hasTrait('imperial') && card.getType() === CardTypes.Character) {
                            context.player.moveCard(card, context.source.location);
                            card.facedown = false;
                            context.player.moveCard(context.source, Locations.DynastyDeck);
                        }
                        this.miyaLibraryPrompt(context, context.player.dynastyDeck.first(4), [], 'Select the card you would like to place on top of your dynasty deck');
                    }`

**Comment by Matthew-Griffith** [comment]
hey have y'all had any issue getting a game node to launch. it work for me in on the develop branch but not the miya library branch. here is the error I get. I want to see what the card is doing in game because the test errors aren't really making sense to me.
```
PS C:\Users\mattg\Documents\projects\js\ringteki> node server/gamenode
module.js:550
    throw err;
    ^

Error: Cannot find module './GameActions/GameActions'
    at Function.Module._resolveFilename (module.js:548:15)
    at Function.Module._load (module.js:475:25)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\mattg\Documents\projects\js\ringteki\server\game\chatcommands.js:2:21)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\mattg\Documents\projects\js\ringteki\server\game\game.js:4:22)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\mattg\Documents\projects\js\ringteki\server\gamenode\gameserver.js:13:14)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Module.require (module.js:597:17)
    at require (internal/module.js:11:18)
    at Object.<anonymous> (C:\Users\mattg\Documents\projects\js\ringteki\server\gamenode\index.js:3:20)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
    at startup (bootstrap_node.js:204:16)
    at bootstrap_node.js:625:3
```

**Comment by jamesoddy** [comment]
@Matthew-Griffith

For anything based upon the `cote` branch you need to run
```
node build\server\gamenode
```
for the game node.  It's because its moved to typescript.

**Comment by jeremylarner** [comment]
Is this ready to merge?

**Comment by Matthew-Griffith** [comment]
I think it is good to be merged. if not let me know I can fix whatever.

**Pull request merged**